dataprofiler/data_readers/graph_data.py

```diff
@@ -0,0 +1,159 @@
+        column_names = cls.csv_column_names(file_path, options)
+        source_keywords = ['source', 'src', 'origin']
+        target_keywords = ['target', 'destination', 'dst']
+        source_index = cls._find_target_string_in_column(column_names, source_keywords)
+        destination_index = cls._find_target_string_in_column(column_names, target_keywords)
+        has_source = True if source_index >= 0 else False
+        has_target = True if destination_index >= 0 else False
+
+        if has_target and has_source:
+            options.update(source_node = source_index)
+            options.update(destination_node = destination_index)
+            options.update(destination_list = target_keywords)
+            options.update(source_list = source_keywords)
+            options.update(column_name = column_names)
+            return True
+
+        return False
+
+    def _format_data_networkx(self):
+        '''
+        Formats the input file into a networkX graph
+        '''
+        source_keywords = ['source', 'src', 'origin']
+        target_keywords = ['target', 'destination', 'dst']
+
+        if self.input_file_path:
+            column_names = self.csv_column_names(self.input_file_path, self.options)
+
+            if self.options.get("source_node") is None or self.options.get("destination_node") is None:
+                source_node = self._find_target_string_in_column(column_names, source_keywords)
+                destination_node = self._find_target_string_in_column(column_names, target_keywords)
+            else:
+                source_node = self.options.get("source_node")
+                destination_node = self.options.get("destination_node")
+
+            # read lines from csv
+            csv_as_list = []
+            with FileOrBufferHandler(self.input_file_path) as csv_file:
+                csv_reader = csv.reader(csv_file, delimiter = self.options.get("delimiter", self.options.get("delimiter", ",")))
+                for row in csv_reader:
+                    row = [item.replace(" ", "") for item in row]
+                    csv_as_list.append(row)
+            
+            # grab list of edges from source/dest nodes
+            list_edges = []
+            list_nodes = set()
+
+            for line in range(1, len(csv_as_list)):
+                # fetch attributes in columns
+                attributes = dict()
+                for column in range(0, len(csv_as_list[0])):
+
+                    if csv_as_list[line][column] is None:
+                        continue
+                    if column is not source_node or column is not destination_node:
+                        attributes[column_names[column]] = (csv_as_list[line][column])
+                        list_edges.append([csv_as_list[line][destination_node], csv_as_list[line][source_node], attributes])
+                    if column is source_node or column is destination_node:
+                        list_nodes.add(csv_as_list[line][column])
```

Comment:
can we add nodes while in here rather than collect all then add? would potentially save memory?